OrbitCore/SymbolHelper.h

```diff
@@ -0,0 +1,35 @@
+#ifndef SYMBOL_HELPER_H_
+#define SYMBOL_HELPER_H_
+#include <string>
+#include <vector>
+
+#include "OrbitModule.h"
+#include "Pdb.h"
+
+class SymbolHelper {
+ public:
+  // TODO check function ordering
+  SymbolHelper();
+  SymbolHelper(std::vector<std::string> collector_symbol_directories,
+               std::vector<std::string> symbols_file_directories)
+      : collector_symbol_directories_(collector_symbol_directories),
+        symbols_file_directories_(symbols_file_directories){};
+
+  bool LoadSymbolsIncludedInBinary(std::shared_ptr<Module> module) const;
+  bool LoadSymbolsCollector(std::shared_ptr<Module> module) const;
+  bool LoadSymbolsUsingSymbolsFile(std::shared_ptr<Module> module) const;
+  void LoadSymbolsFromDebugInfo(std::shared_ptr<Module> module,
+                                const ModuleDebugInfo& module_info) const;
+  void FillDebugInfoFromModule(std::shared_ptr<Module> module,
+                               ModuleDebugInfo& module_info) const;
+
+  std::vector<std::string> GetSymbolsFileDirectories() const {
+    return symbols_file_directories_;
+  }
+
+ private:
+  const std::vector<std::string> collector_symbol_directories_;
+  std::vector<std::string> symbols_file_directories_;
```

Comment:
const?

Reply:
You probably didn't make it const because you modify this member in the constructor, right? You can wrap your constructor code in a free function and initialize this member from the return type of this free function.

Reply:
I was actually looking for a way to do this, thank you!

Reply:
Cool. Last small thing. Can you put your free function in an anonymous namespace?
Otherwise it will still be visible to the linker and if somebody else happens to use the same
name somewhere else it will result in a ODR issue.

Reply:
makes sense to me, thank you!